src/AEG/AtomNode.ts

```diff
@@ -12,29 +12,74 @@ export class AtomNode {
     /**
      * The rectangle signifying the boundary box of this node.
      */
-    rect: Rectangle;
+    private rect: Rectangle;
 
     /**
      * The string value of the proposition represented by this node.
      */
-    identifier: string;
+    private identifier: string;
 
     /**
      * The point the atom is initially placed.
      */
-    origin: Point;
+    private origin: Point;
 
     /**
      * Construct an atom node with given boundary box and proposition.
-     * @param rect The rectangle to be set as the boundary box of this node.
-     * @param val The value of the proposition represented by this node.
+     * @param rect (Required) The rectangle to be set as the boundary box of this node.
+     * @param val (Required) The value of the proposition represented by this node.
      */
-    public constructor(val?: string, origin?: Point, rect?: Rectangle) {
+    public constructor(val: string, origin?: Point, rect?: Rectangle) {
         this.rect = rect ?? new Rectangle();
-        this.identifier = val ?? "";
+        this.identifier = val;
         this.origin = origin ?? new Point();
     }
 
+    /**
+     * Accessor to get the bounding rectangle of the Atom Node.
+     * @returns The bounding rectangle of this Atom Node
+     */
+    public get Rectangle(): Rectangle {
+        return this.rect;
+    }
+
+    /**
+     * Modifier to set the bounding rectangle of the Atom Node.
+     */
+    public set Rectangle(rect: Rectangle) {
+        this.rect = rect;
+    }
+
+    /**
+     * Accessor to get the identifier of the Atom Node.
+     * @returns The identifier of this Atom Node
+     */
+    public get Identifier(): string {
+        return this.identifier;
+    }
+
+    /**
+     * Modifier to set the identifier of the Atom Node
+     */
+    public set Identifier(identifier: string) {
+        this.identifier = identifier;
+    }
+
+    /**
+     * Accessor to get the origin (top left) vertex of the bounding rectangle of the Atom Node.
+     * @returns The origin of the bounding rectangle
+     */
+    public get Origin(): Point {
+        return this.origin;
+    }
```

Comment:
Does the capitalization of these setter and getter names interfere with our ability to do .rectangle and .origin? I'm under the impression that its best practice to have them use the same name as the member variable but could be wrong.